snuba/clusters/storage_sets.py

```diff
@@ -0,0 +1,11 @@
+from typing import Mapping, Set
+
+
+STORAGE_SETS: Mapping[str, Set[str]] = {
```

Comment:
I think this relation storageset -> storage is going in the wrong direction.If you make the storageset define a set of storages you are building an m..n relationship . It is only by convention at runtime that you are enforcing that is is actually a 1..n.
If you let the storage instance point to the storage_set you do not have this problem. There structurally it would be impossible to miss a storage or to assign a storage to multiple storage sets, it also comes with the bonus that you do not need the storage_key anymore.

Is there any use case where you cannot go that way ?

Reply:
There was not really a reason except that I thought it'd be slightly easier to see the list of storage sets required to be registered. I've changed the structure of this to go the other way.